### ISSUE TITLE
Linter Syntax validation not refreshing after editing file

### ISSUE THREAD

**Issue by vyacheslav31** [hashicorp/vscode-terraform#1806]
### Extension Version

v2.32.1

### VS Code Version

Version: 1.91.1
Commit: f1e16e1e6214d7c44d078b1f0607b2388f29d729
Date: 2024-07-09T22:07:46.768Z (2 wks ago)
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Darwin arm64 23.5.0

### Operating System

macOS Sonoma Version 14.5 (23F79)

### Terraform Version

Terraform v1.9.2 on darwin_arm64

### Steps to Reproduce

1. Install VSCode.
2. Install Terraform and HashiCorp Terraform Extension
3. Create a .tf file
4. Create write a valid terraform config like: 

```HCL
provider "google" {
  project                     = local.project_id
  region                      = local.region
  impersonate_service_account = var.impersonation_service_account
}
```

5. Add a random letter to the end of local.region => local.regiona
6. Remove the extra letter
7. Linter should refresh and now say the config is valid.
8. Error still exists.

### Expected Behavior

Linter should correctly not display errors anymore once the terraform configuration is valid.

### Actual Behavior

Linter still shows errors.
<img width="676" alt="image" src="https://github.com/user-attachments/assets/f8d557dc-579d-4325-ada2-79e128183e92">

### Project Structure

```sh
.
├── environments
│   ├── dev
│   │   └── main.tf
│   ├── prod
│   │   └── main.tf
│   └── uat
│       └── main.tf
└── modules
    └── project
        ├── artifact-registry.tf
        ├── build.tf
        ├── cloudrun.tf
        ├── locals.tf
        ├── providers.tf
        └── variables.tf
```

### Help Wanted

- [ ] I'm interested in contributing a fix myself

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by dbanck** [comment]
Hi @vyacheslav31! Thanks for the report, I'm able to reproduce the issue and am looking for a fix.

**Comment by dbanck** [comment]
We have just released a new version of vscode-terraform: [2.32.2](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.32.2) that fixes the reported issue. The update should show up automatically in VS Code.

Thanks again for the report @vyacheslav31
